src/ast/document-descriptor.ts

```diff
@@ -146,6 +148,9 @@ export class Document implements Feature {
   getByKind(kind: 'import'): Set<Import>;
   getByKind(kind: string): Set<Feature>;
   getByKind(kind: string): Set<Feature> {
+    if (this._featuresByKind) {
```

Comment:
I would make these caches, so that they're populated as the queries come in. This will discourage callers from holding references to things, if the second query is always fast.